src/BudgetManager/BudgetManager.API/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,54 @@
+﻿using BudgetManager.CQRS.Commands.CategoryCommands;
+using BudgetManager.CQRS.Queries.CategoryQueries;
+using BudgetManager.SDK.DTO.CategoryDTOs;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace BudgetManager.API.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class CategoryController : Controller
+    {
+        private readonly IMediator _mediator;
+        public CategoryController(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        [HttpGet("[action]")]
+        public async Task<IActionResult> GetAll(CancellationToken cancellationToken)
+        {
+            var response = await _mediator.Send(new GetCategoriesQuery(), cancellationToken);
+            return Ok(response);
+        }
+
+        [HttpGet("[action]/{id}")]
+        public async Task<IActionResult> GetById(Guid id, CancellationToken cancellationToken)
+        {
+            var response = await _mediator.Send(new GetOneCategoryQuery(id), cancellationToken);
+            return Ok(response);
+        }
+
+        [HttpPost("[action]")]
+        public async Task<IActionResult> InsertOne(AddCategoryDTO category, CancellationToken cancellationToken)
+        {
+            var response = await _mediator.Send(new AddCategoryCommand(category), cancellationToken);
+            return Ok(response);
+        }
+
+        [HttpPut("[action]")]
+        public async Task<IActionResult> UpdateOne(UpdateCategoryDTO category, CancellationToken cancellationToken)
+        {
+            var response = await _mediator.Send(new UpdateCategoryCommand(category), cancellationToken);
+            return Ok(response);
+        }
+
+        [HttpDelete("[action]/{id}")]
+        public async Task<IActionResult> DeleteOne(Guid id, CancellationToken cancellationToken)
+        {
+            var response = _mediator.Send(new DeleteCategoryCommand(id), cancellationToken);
+            return Ok(response);
```

Comment:
If category deletion fails it is going to return ok response.
![image](https://user-images.githubusercontent.com/92161964/180817100-af16744f-885c-4903-9e5d-9c1f124b3acc.png)
Should probably return a bad request.

Reply:
> If category deletion fails it is going to return ok response. ![image](https://user-images.githubusercontent.com/92161964/180817100-af16744f-885c-4903-9e5d-9c1f124b3acc.png) Should probably return a bad request. Similar situation with some other controller operations.

Will fix it